src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerTests.cs

```diff
@@ -40,20 +40,309 @@ public async Task GetInvitationsByCommPkgNo_AsViewer_ShouldGetInvitations()
             Assert.AreEqual(KnownTestData.InvitationDescription, invitation.Description);
         }
 
+        [TestMethod]
+        public async Task SignPunchOut_AsPlanner_ShouldSignPunchOut()
+        {
+            // Arrange
+            var invitationToSignId = await InvitationsControllerTestsHelper.CreateInvitationAsync(
+                UserType.Planner,
+                TestFactory.PlantWithAccess,
+                "InvitationForSigningTitle",
+                "InvitationForSigningDescription",
+                InvitationLocation,
+                DisciplineType.DP,
+                _invitationStartTime,
+                _invitationEndTime,
+                _participantsForSigning,
+                _mcPkgScope,
+                null
+            );
+
+            var invitation = await InvitationsControllerTestsHelper.GetInvitationAsync(
+                UserType.Viewer,
+                TestFactory.PlantWithAccess,
+                invitationToSignId);
+
+            var participantPerson = invitation.Participants
+                .Single(p => p.Organization == Organization.TechnicalIntegrity).Person;
+
+            // Act
+            if (participantPerson.Person.Id != null)
+            {
+                await InvitationsControllerTestsHelper.SignPunchOutAsync(
+                    UserType.Signer,
+                    TestFactory.PlantWithAccess,
+                    invitationToSignId,
+                    (int)participantPerson.Person.Id,
```

Comment:
participantPerson.Person.Id.Value

Reply:
really nice catch, not sure why I didn't see this myself when doing all these casts and nullchecks